tests/run_tests.py

```diff
@@ -351,28 +357,37 @@ def run_system(config_file, vic_exe, test_data_dir, out_dir, driver):
                                                         expectation))
 
             # -------------------------------------------------------- #
-            # check output files
-            if test_dict['check']:
-                if 'complete' in test_dict['check']:
-                    start = None
-                    end = None
-
-                for fname in glob.glob(os.path.join(dirs['results'], '*')):
-                    df = read_vic_ascii(fname, header=True)
-                    # do numeric tests
-
-                    # check that each dataframe includes all timestamps
+            # check output files (different tests depending on driver)
+            if driver == "classic":
```

Comment:
Would it be better to reorder the if statements, say, put `if test_dict['check']` as the highest level, and then check for each possible checks, and then the driver options if needed? Something like this:
```
if test_dict['check']:
        if 'complete' in test_dict['check']:
                do something (maybe another if statement for classic or image driver, if needed)
        if 'nonan' in test_dict['check']:
                do something
```

To me it would be cleaner, and another reason I'm suggesting this is because for some more complicated tests (e.g., the restart tests), maybe only the reading data part is different for different drivers, but the other checking process would be the same. In those cases, it makes more sense to put the distinguishing of drivers in the most inner layer, so that the same checking function can be called.